crates/alerter/src/main.rs

```diff
@@ -211,22 +251,34 @@ impl Alerter {
                 break;
             }
         }
+
         tracing::debug!("found no fulfilled orders");
         Ok(())
     }
 
-    fn alert(&self, order: &Order) {
-        tracing::error!(
-            "No orders have been settled in the last {} seconds even though order {} is solvable and has a price that allows it to be settled according to 0x.",
-            self.config.time_without_trade.as_secs(),
-            order.uid,
-        );
-    }
-
     pub async fn update(&mut self) -> Result<()> {
         self.update_open_orders().await?;
+        let matchable_orders = self.matchable_orders().await?;
+
+        let counts = matchable_orders.iter().fold(
+            OrderClass::VARIANTS
```

Comment:
I had to do the same thing recently and didn't think about `.fold()`. I still don't love it but at least it's self contained. 👍